src/SkyDrop.Droid/Helper/AndroidUtil.cs

```diff
@@ -47,21 +39,26 @@ public static int DpToPx(int dp)
         /// </summary>
         public static string GetFileName(Context context, Android.Net.Uri uri)
         {
-            if (uri.LastPathSegment.Contains("."))
-            {
-                //this is a "file"
-                return uri.LastPathSegment;
-            }
+            if (context == null)
+                throw new ArgumentNullException(nameof(context));
 
-            //this is an "image"
-
-            // The query, because it only applies to a single document, returns only
-            // one row. There's no need to filter, sort, or select fields,
-            // because we want all fields for one document.
-            Android.Database.ICursor cursor = context.ContentResolver.Query(uri, null, null, null, null, null);
-            var displayName = "";
+            string displayName;
+            Android.Database.ICursor cursor = null;
             try
             {
+                if (uri.LastPathSegment.Contains("."))
+                {
+                    //this is a "file"
+                    return uri.LastPathSegment;
+                }
+
+                //this is an "image"
+
+                // The query, because it only applies to a single document, returns only
+                // one row. There's no need to filter, sort, or select fields,
+                // because we want all fields for one document.
+                cursor = context.ContentResolver.Query(uri, null, null, null, null, null);
+                
                 // moveToFirst() returns false if the cursor has 0 rows. Very handy for
```

Comment:
This method was just causing me some IDE issue too, just refactored the try/catch a bit as well